### PR TITLE
feat(addon-doc): support lazy component for preview

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/addon-doc/components/example/example.component.ts [hunk 1]
```diff
@@ -78,13 +84,27 @@ export class TuiDocExampleComponent {
             map(value => this.processContent(value)),
         );
 
+    protected readonly lazyComponent$ = this.lazyLoader$$.pipe(
+        switchMap(async lazyImport =>
```
Comment (waterplea) [inline projects/addon-doc/components/example/example.component.ts:88]: Maybe `switchAll` + `map` so we don't mix RxJS and async.then? If I remember correctly what switchAll does :)

File: projects/demo/src/modules/components/avatar/index.html [hunk 2]
```diff
@@ -7,62 +7,59 @@
         <tui-doc-example
             id="content"
             heading="Content types"
-            [content]="example1"
+            [component]="example1"
```
Comment (waterplea) [inline projects/demo/src/modules/components/avatar/index.html:10]: Maybe a pipe like `tuiExample`?

File: projects/addon-doc/components/example/example.component.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-doc/components/example/example.component.ts]: fixed

File: projects/demo/src/modules/components/avatar/index.html (hunk 2 above)
Comment (splincode) [reply projects/demo/src/modules/components/avatar/index.html]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**